### PR TITLE
feat(custom generators): added '--custom-generator' parameter fixes #237

### LINKED ISSUE #237: [missing feature] custom generators on classpath are not visible
# Custom generators are not visible to openapi-generator

Custom generators can be shipped as `.jar` file with codegen config and templates. This works fine with `maven`/`gradle` plugins, just add the `.jar` with a custom generator to dependencies. However, this will not work with `openapi-generator-cli` because [generator .jar is called using `-jar` option](https://github.com/OpenAPITools/openapi-generator-cli/blob/master/apps/generator-cli/src/app/services/pass-trough.service.ts#L89). When java is started with `-jar` option, [then `-cp` (classpath) option is ignored](https://docs.oracle.com/javase/7/docs/technotes/tools/solaris/java.html#jar).

A solution to this could be use of class path and main class explicitly
```
  java -cp "${userProvidedClasspath}:${this.versionManager.filePath()}" org.openapitools.codegen.OpenAPIGenerator ...
```
This will work the same as with `-jar` option, but provides the user with an option to declare custom generator.

### Steps to Reproduce

1. Make a[ new custom generator](https://openapi-generator.tech/docs/new-generator)
2. Add script to `package.json` with `JAVA_OPTS='-cp custom-generator.jar' openapi-generator-cli list`
2. You will not see the generator because `-cp` option is ignored

### Expected behavior
A mechanism to add custom `.jar` files to classpath. This will make it possible to ship custom generator as a `.jar` file.

I can make a pull request with this fix if you approve this idea. `PassThroughService` can simply check if class path extensions are provided. If not, then run generator with `-jar` option. Otherwise, assemble class path and run main class directly.

### PR DESCRIPTION
**Pull request by mcaden**
usage:

`--custom-generator=path/to/my/generator.jar` 

It handles the difference in the classpath delimiter on windows and other systems.

Should fix #237 

Tested in Windows and Ubuntu.

### CODE REVIEW

**Comment by SardarNL** [comment]
Does it work on your machine? I tried:
```
npm run build
npm run cli -- --custom-generator=/path/to/extensions.jar list
```

I get `error: unknown option '--custom-generator=...'`

**Comment by mcaden** [comment]
This works:

`npm run cli -- list --custom-generator=/path/to/extensions.jar`

The difference is simply that "list" is before the `--custom-generator`

**Comment by SardarNL** [comment]
Indeed, it works!

I managed to do the same by simply cloning this repo and building our own cli wrapper. But the official solution is way better.

**Comment by mcaden** [comment]
@kay-schecker Please let me know if there's anything else I need to do in order to make this eligible for merge.

**Review by kay-schecker — APPROVED** [review]

**Comment by kay-schecker** [comment]
Hi, please fix the linting errors

**Comment by mcaden** [comment]
@kay-schecker Ah, fixed. I'm too used to that being a part of the build/commit process and didn't run lint separately.

File: apps/generator-cli/src/app/mocks/passthrough-command.mock.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export class PassthroughCommandMock {
+  name = jest.fn().mockReturnValue(this._name);
+  options: { [key: string]: string } = {};
```
Comment (kay-schecker) [inline apps/generator-cli/src/app/mocks/passthrough-command.mock.ts:3]: Could you use `Record<string, string>` instead, please?

Comment (mcaden) [reply apps/generator-cli/src/app/mocks/passthrough-command.mock.ts]: done

**Pull request merged**